lms/views/dashboard/api/course.py

```diff
@@ -8,12 +12,49 @@
     APICourses,
     CourseMetrics,
 )
-from lms.models import RoleScope, RoleType
+from lms.models import Course, RoleScope, RoleType
 from lms.security import Permissions
 from lms.services.h_api import HAPI
 from lms.services.organization import OrganizationService
+from lms.validation._base import PyramidRequestSchema
 from lms.views.dashboard.base import get_request_course, get_request_organization
 
+PAGINATION_LIMIT = 100
+"""Maximum numbr of items to return in paginated endpoints"""
+
+
+class ListCoursesSchema(PyramidRequestSchema):
+    location = "query"
+
+    organization = fields.Str()
+    """Only list courses that belong to this organization"""
+
+    limit = fields.Integer(required=True)
+    """Maximum number of items to return."""
+
+    cursor = fields.Str()
+    """Position to return the elements from. This correponds to the `next` value in the repsonse."""
+
+
+def _next_parameter(courses: list[Course]) -> str:
+    """Build the `next` parameter for a list of courses.
+
+    We'll sort courses by both name and id, expose both as a tuple.
+    """
+    last_element = courses[-1]
+
+    cursor_data = (last_element.lms_name, last_element.id)
+    return (
+        base64.urlsafe_b64encode(json.dumps(cursor_data).encode("utf-8"))
```

Comment:
base64-ing a json list is probably the most immediate solution to encode two values.